### PR TITLE
Add solution to cses 1093 two sets

### PR DESCRIPTION
**Pull request by TruaShamu**

### CODE REVIEW

File: solutions/cses-1093.mdx [hunk 1]
```diff
@@ -0,0 +1,70 @@
+---
+id: cses-1093
+source: CSES
+title: Two Sets II
+author: Sofia Yang
+---
+[Codeforces Editorial](https://codeforces.com/blog/entry/70018)
```
Comment (bqi343) [inline solutions/cses-1093.mdx:7]:
```suggestion
[Official Editorial](https://codeforces.com/blog/entry/70018)
```

to be consistent with the other solutions. looks like two other solutions w/ "Codeforces Editorial" were already merged in - those should be changed as well.

File: solutions/cses-1093.mdx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+import java.util.*;
+
+public class cses1093 {
+    static final int MOD = (int) 1e9 + 7;
+
+    public static void main(String[] args) {
+        Kattio io = new Kattio();
+        int N = io.nextInt();
+        // The total sum of all N elements.
+        long totalSum = (long) (N * (N + 1)) / 2;
```
Comment (bqi343) [inline solutions/cses-1093.mdx:30]: why not an `int`?

**Review by bqi343 — CHANGES_REQUESTED** [review]

File: solutions/cses-1093.mdx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+import java.util.*;
+
+public class cses1093 {
+    static final int MOD = (int) 1e9 + 7;
+
+    public static void main(String[] args) {
+        Kattio io = new Kattio();
+        int N = io.nextInt();
+        // The total sum of all N elements.
+        int totalSum = (N * (N + 1)) / 2;
```
Comment (bqi343) [inline solutions/cses-1093.mdx:30]:
```suggestion
        int totalSum = N * (N + 1) / 2;
```

File: solutions/cses-1093.mdx [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+        // This cannot be partitioned into 2 equal subsets.
+        if (totalSum % 2 != 0) {
+            System.out.println(0);
```
Comment (bqi343) [inline solutions/cses-1093.mdx:34]: Why `System.out.println` here but `io.println` later on?

**Review by bqi343 — CHANGES_REQUESTED** [review]

File: solutions/cses-1093.mdx [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+            System.exit(0);
+        }
+        totalSum /= 2;
+
+        /*
+         * DP[i][j] is the number of ways to make sum j with the first i elements.
+         * The range of 'idx' is from 0 to N-1 because if all elements are used,
+         * the subsets will be double counted.
+         */
+        int[][] dp = new long[N][totalSum + 1];
```
Comment (dongliu0426) [inline solutions/cses-1093.mdx:44]:
```suggestion
        int[][] dp = new int[N][totalSum + 1];
```

File: solutions/cses-1093.mdx [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+
+    public static void main(String[] args) {
+        Kattio io = new Kattio();
+        int N = io.nextInt();
+        // The total sum of all N elements.
+        int totalSum = N * (N + 1) / 2;
+
+        // This cannot be partitioned into 2 equal subsets.
+        if (totalSum % 2 != 0) {
+            io.println(0);
```
Comment (bqi343) [inline solutions/cses-1093.mdx:34]: code no longer prints anything when output is 0 D:

File: solutions/cses-1093.mdx [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+
+```java
+import java.io.*;
+import java.util.*;
+
+public class cses1093 {
+    static final int MOD = (int) 1e9 + 7;
+
+    public static void main(String[] args) {
+        Kattio io = new Kattio();
```
Comment (bqi343) [inline solutions/cses-1093.mdx:27]: include `Kattio` as a codesnip

**Review by bqi343 — CHANGES_REQUESTED** [review]

**Comment by SansPapyrus683** [comment]
also indent w/ tabs?

**Comment by TruaShamu** [comment]
Ok, I fixed it.

**Review by bqi343 — APPROVED** [review]

File: solutions/cses-1093.mdx [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+
+<LanguageSection>
+
+<JavaSection>
+
+```java
+import java.io.*;
+import java.util.*;
+
+public class cses1093 {
```
Comment (SansPapyrus683) [inline solutions/cses-1093.mdx:23]: TwoSets instead of cses1093?
other than that very cool

**Review by SansPapyrus683 — APPROVED** [review]

**Pull request merged**